lib/src/idx/trees/mtree.rs

```diff
@@ -211,28 +211,39 @@ impl KnnResultBuilder {
 		let pr = PriorityResult(dist);
 		match self.priority_list.entry(pr) {
 			Entry::Vacant(e) => {
+				self.docs |= docs;
 				e.insert(docs.clone());
 			}
 			Entry::Occupied(mut e) => {
-				let d = e.get_mut();
-				for doc in docs {
-					d.insert(doc);
-				}
+				*e.get_mut() |= docs;
+				self.docs |= docs;
 			}
 		}
+
+		#[cfg(debug_assertions)]
+		debug!("KnnResult add - docs: {} - total: {}", docs.len(), self.docs.len());
+
 		// Do possible eviction
 		let docs_len = self.docs.len();
 		if docs_len > self.knn {
 			if let Some((_, d)) = self.priority_list.last_key_value() {
 				if docs_len - d.len() >= self.knn {
-					self.priority_list.pop_last();
+					if let Some((_, evicted_docs)) = self.priority_list.pop_last() {
+						self.docs -= evicted_docs;
+					}
```

Comment:
This was missing for a proper eviction queue.  The evicted docs were not remove from `self.docs`